pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -62,9 +60,11 @@ pub struct InMemoryLayerInner {
     ///
     segsizes: BTreeMap<Lsn, u32>,
 
-    /// True if freeze() has been called on the layer, indicating it no longer
-    /// accepts writes.
-    frozen: bool,
+    /// This layer is in the process of being replaced, no writes are allowed.
+    writeable: bool,
```

Comment:
I suggest rewording the comment so that it matches the variable name.

This variable should help to implement multiple `frozen` layers:
1. When some layer is dropped - mark it as non-writeable.
2. When searching for a layer to write, get the latest open. If it is non-writeable, just create a new open layer.
I think that should work.

Reply:
I see. Would the checkpointer thread still be responsible for flushing these non-writeable layers out to disk?